rust/datafusion/src/physical_plan/expressions.rs

```diff
@@ -2499,4 +2574,56 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn is_null_op() -> Result<()> {
+        let schema = Schema::new(vec![Field::new("a", DataType::Utf8, true)]);
+        let a = StringArray::from(vec![Some("foo"), None]);
+        let batch = RecordBatch::try_new(Arc::new(schema.clone()), vec![Arc::new(a)])?;
+
+        // expression: "a is null"
+        let expr = is_null(col("a")).unwrap();
+        let result = expr.evaluate(&batch)?;
+
+        let expected = vec![false, true];
+
+        let result = result
+            .as_any()
+            .downcast_ref::<BooleanArray>()
+            .expect("failed to downcast to BooleanArray");
+
+        assert_eq!(result.len(), expected.len());
+
+        for i in 0..expected.len() {
+            assert_eq!(result.value(i), expected[i]);
+        }
+
+        Ok(())
+    }
+
+    #[test]
+    fn is_not_null_op() -> Result<()> {
+        let schema = Schema::new(vec![Field::new("a", DataType::Utf8, true)]);
+        let a = StringArray::from(vec![Some("foo"), None]);
+        let batch = RecordBatch::try_new(Arc::new(schema.clone()), vec![Arc::new(a)])?;
+
+        // expression: "a is null"
+        let expr = is_not_null(col("a")).unwrap();
+        let result = expr.evaluate(&batch)?;
+
+        let expected = vec![true, false];
+
+        let result = result
+            .as_any()
+            .downcast_ref::<BooleanArray>()
+            .expect("failed to downcast to BooleanArray");
+
+        assert_eq!(result.len(), expected.len());
+
```

Comment:
If we use
```
let expected = &BooleanArray::from(vec![true, false]);
```

then we can use 

```
assert_eq!(result, expected);
```

here, which may be more expressive of our intent.